pkg/mkimage-raw-efi/make-raw

```diff
@@ -232,20 +232,17 @@ do_persist() {
 
 do_usb_conf() {
     eval local SEC_START="\$$1"
-    # usb_conf partition is expected to be the only one and takes all available space,
-    # but at the same time be no bigger than 6G or 12582912 in sector size
     # shellcheck disable=SC2155
     local SEC_END=$(sgdisk -E "$IMGFILE")
     local NUM_PART=$(( PART_OFFSET + 1 ))
     local PART_TYPE=EBD0A0A2-B9E5-4433-87C0-68B6B72699C7
-
-    [ $(( SEC_END - SEC_START - 12582912 )) -gt 0 ] && SEC_END=12582912
+    local FAT_SIZE=$(( SEC_END - SEC_START ))
 
     sgdisk --new "$NUM_PART:$SEC_START:$SEC_END" \
            --typecode="$NUM_PART:$PART_TYPE" \
            --change-name="$NUM_PART:DevicePortConfig" "$IMGFILE"
 
-    mformat -i "${IMGFILE}@@$(( SEC_START * 512 ))" -h 1 -t 1 -s $(( SEC_END - SEC_START )) -l EVEDPC ::
+    mformat -i "${IMGFILE}@@$(( SEC_START * 512 ))" -h $(( FAT_SIZE / 65535 + 1 )) -t 1 -s 65535 -l EVEDPC ::
```

Comment:
This part confused me. All good cleanup, turning the differential into `FAT_SIZE`, etc. However, the calculation changes threw me off. Originally we had to have `SEC_END` not pass `12582912` (I don't know why, just accepting it); does that no longer matter? We also fixed the sector count to 65535, as opposed to the differential `SEC_END - SEC_START` from before. The `-h` shouldn't matter all that much (heads? anything actually pay attention to that nowadays?).

Reply:
That's the quirks of different versions of mtools. The ones that we used before (from alpine 3.8) did have this arbitrary limit of 12582912 (which -- just as yourself -- I just accepted). The new ones (from alpine 3.10) actually have the sector limit at 65535 (dunno why) BUT they now accept -h and it seems that the overall size of the vfat partition gets calculated as h*s*t -- so that's what I'm using now.

I tested it with Mac OS X -- and it seems that the filesystem gets mounted without many issues.

But this also points at the fact that mtools are slow, quirky and not all that well maintained, we need to start using something like mkfs.vfat here -- but the problem is that only unreleased version supports offset (same as @@ setting in my code). So I need to build it from source if I want to use it.

That's also definitely on my TODO list.